src/main/java/no/fdk/dataservicecatalog/service/parser/Parser.java

```diff
@@ -0,0 +1,49 @@
+package no.fdk.dataservicecatalog.service.parser;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import no.fdk.dataservicecatalog.dto.shared.apispecification.ApiSpecification;
+import no.fdk.dataservicecatalog.exceptions.ParseException;
+
+public interface Parser {
+
+    enum ApiType {
+        OPENAPI("openapi"),
+        SWAGGER("swagger");
+
+        public final String label;
+
+        ApiType(String label) {
+            this.label = label;
+        }
+
+    }
+
+    boolean canParse(String spec);
+
+    ApiSpecification parse(String spec) throws ParseException;
+
+    static boolean isValidSwaggerOrOpenApiV3(String spec, ApiType apiType, String minimalVersion) {
+        try {
+            JsonElement element = new JsonParser().parse(spec);
```

Comment:
Can we avoid parsing JSON to determine version?

Reply:
That would certainly be a lot better!

Reply:
In a separate PR

Reply:
zenhub -> new issue